apps/web-giddh/src/app/services/receipt.service.ts

```diff
@@ -289,4 +308,44 @@ export class ReceiptService implements OnInit {
             .replace(':purchaseRecordUniqueNumber', purchaseRecordUniqueName)
         ).pipe(catchError((e) => this.errorHandler.HandleCatch<Voucher, any>(e)));
     }
+
+    /**
+     * Fetches all the advance receipts
+     *
+     * @param {GetAllAdvanceReceiptsRequest} requestObject Request object
+     * @returns {Observable<BaseResponse<any, GetAllAdvanceReceiptsRequest>>} Observable to carry out further operations
+     * @memberof ReceiptService
+     */
+    public getAllAdvanceReceipts(requestObject: GetAllAdvanceReceiptsRequest): Observable<BaseResponse<any, GetAllAdvanceReceiptsRequest>> {
+        const companyUniqueName = this._generalService.companyUniqueName;
+        return this._http.post(
+            `${this.config.apiUrl}${RECEIPT_API.GET_ALL_ADVANCE_RECEIPTS}`
+                .replace(':companyUniqueName', encodeURIComponent(companyUniqueName))
+                .replace(':sortBy', requestObject && requestObject.sortBy ? requestObject.sortBy : '')
+                .replace(':sort', requestObject && requestObject.sort ? requestObject.sort : '')
+                .replace(':page', (requestObject && requestObject.page) ? String(requestObject.page) : '')
+                .replace(':count', requestObject && requestObject.count ? String(requestObject.count) : '')
+                .replace(':from', requestObject && requestObject.from ? requestObject.from : '')
+                .replace(':to', requestObject && requestObject.to ? requestObject.to : ''),
+            requestObject
+        ).pipe(catchError((error) => this.errorHandler.HandleCatch<any, GetAllAdvanceReceiptsRequest>(error)));
+    }
+
+    /**
+     * Fetches the summary advance receipts based on date filters
+     *
+     * @param {AdvanceReceiptSummaryRequest} requestObject Request object
+     * @returns {Observable<BaseResponse<any, AdvanceReceiptSummaryRequest>>} Observable to carry out further operations
+     * @memberof ReceiptService
+     */
+    public fetchSummary(requestObject: AdvanceReceiptSummaryRequest): Observable<BaseResponse<any, AdvanceReceiptSummaryRequest>> {
+        const companyUniqueName = this._generalService.companyUniqueName;
```

Comment:
Don't use companyUniqueName from general service. It will come wrong if we will switch company.